csunplugged/utils/BaseResourceGenerator.py

```diff
@@ -76,3 +79,29 @@ def process_requested_options(self, requested_options):
                 values[i] = update_value
             requested_options.setlist(option, values)
         return requested_options
+
+    def thumbnail(self):
```

Comment:
I'd suggest you make this a function `generate_thumbnail(self, path)` which will actually write out the image file. The 'override' mechanism can then become the subclass overloading this function, to save a pregenerated PNG rather than dealing with pdf generation and conversion to png. The subclass implementation can choose to ignore the requested options, and generate the same png for every combination, if it so chooses.

This would mean the logic in lines 59-72 in makeresourcethumbnails.py will move into this function.

Reply:
Would this mean the same should happen for making a resource PDF?